atomic_reactor/build.py

```diff
@@ -54,37 +53,35 @@ def _ensure_not_built(self):
 
 
 class BuildResult(object):
-    def __init__(self, command_result, image_id=None):
-        """ when build fails, image_id is None """
-        self.command_result = command_result
+    def __init__(self, logs=None, fail_reason=None, image_id=None):
+        """
+        :param logs: iterable of log lines (without newlines)
+        :param fail_reason: str, description of failure or None if successful
+        :param image_id: str, ID of built container image
+        """
+        assert fail_reason == None or bool(fail_reason), \
```

Comment:
`== None` is better written as `is None`, but see https://github.com/projectatomic/atomic-reactor/pull/647/files#r102438121.